components/dashboard/src/projects/Prebuilds.tsx

```diff
@@ -248,6 +248,8 @@ export function prebuildStatusLabel(prebuild?: PrebuildWithStatus) {
             return (<span className="font-medium text-blue-500 uppercase">running</span>);
         case "aborted":
             return (<span className="font-medium text-gray-500 uppercase">canceled</span>);
+        case "failed":
```

Comment:
Drive-by comment: You probably want to handle `"failed"` in `prebuildStatusIcon` below as well (there is already a `StatusFailed` icon you can re-use).

Reply:
As the number of error cases grows, it becomes harder to enumerate them. One possible way to simplify the API UX is to have a single `error` state and extend it with a `cause` which is set to a computer readable reason for the failure - TimedOut, RateLimited, FailedToStart, etc. The error message then becomes a human readable string to show in the UI etc.

Reply:
Good point @easyCZ. Alternatively, I think we could also special-case the few non-error cases, then consider "all the rest" as errors?

Reply:
Depends how far we want to go. Some failures are due to user issues, some can be due to system issues. I think the it mostly depends if we control all the `clients` which will be interpreting the status (Dashboard, etc) or if some of the clients are/will be out of our control (implementing against an API). In the former case, it doesn't matter much as we can largely reach a shared understanding of what is an error and what isn't. But if an external frontend/server needs to `switch` on these, it may be less obivous to them with a large number of cases.

I think this is probs not something we need/should decide now and I don't think it blocks the change.